module/kvmfr.c

```diff
@@ -139,11 +125,30 @@ static void release_kvmfrbuf(struct dma_buf * buf)
 
 static int mmap_kvmfrbuf(struct dma_buf * buf, struct vm_area_struct * vma)
 {
+  struct kvmfrbuf * kbuf = (struct kvmfrbuf *)buf->priv;
+  unsigned long size = vma->vm_end - vma->vm_start;
+  unsigned long offset = vma->vm_pgoff << PAGE_SHIFT;
+
+  if ((offset + size > (kbuf->pagecount << PAGE_SHIFT)) || (offset + size < offset))
+    return -EINVAL;
+
   if ((vma->vm_flags & (VM_SHARED | VM_MAYSHARE)) == 0)
     return -EINVAL;
-  vma->vm_ops          = &kvmfr_vm_ops;
-  vma->vm_private_data = buf->priv;
-  return 0;
+
+  switch (kbuf->kdev->type)
+  {
+    case KVMFR_TYPE_PCI:
+    {
+      unsigned long pfn = virt_to_phys(kbuf->kdev->addr + kbuf->offset + offset) >> PAGE_SHIFT;
```

Comment:
`virt_to_phys` returns `phys_addr_t`, which while `remap_pfn_range` accepts an `unsigned long`, would seem more correct.

Reply:
I missed the shift, ignore this.